File: app/src/main/java/com/example/secondtask/BookAdapter.java
package com.example.secondtask;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

public class BookAdapter extends RecyclerView.Adapter<BookAdapter.BookViewHolder> {

    private List<Book> bookList;

    public BookAdapter(List<Book> bookList) {
        this.bookList = bookList;
    }

    @NonNull
    @Override
    public BookViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.book_layout, parent, false);
        return new BookViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull BookViewHolder holder, int position) {
        Book book = bookList.get(position);
        holder.titleView.setText(book.getTitle());
        holder.authorView.setText(book.getAuthor());
        holder.descriptionView.setText(book.getDescription());
        holder.imageView.setImageResource(book.getImageResource());
    }

    @Override
    public int getItemCount() {
        return bookList.size();
    }

    static class BookViewHolder extends RecyclerView.ViewHolder {
        TextView titleView;
        TextView authorView;
        TextView descriptionView;
        ImageView imageView;

        public BookViewHolder(@NonNull View bookView) {
            super(bookView);
            titleView = bookView.findViewById(R.id.title);
            authorView = bookView.findViewById(R.id.author);
            descriptionView = bookView.findViewById(R.id.description);
            imageView = bookView.findViewById(R.id.image);
        }
    }
}

File: app/src/main/java/com/example/secondtask/BookAdapter.kt
package com.example.secondtask

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

class BookAdapter(private val bookList: List<Book>) : RecyclerView.Adapter<BookAdapter.BookViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BookViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.book_layout, parent, false)
        return BookViewHolder(view)
    }

    override fun onBindViewHolder(holder: BookViewHolder, position: Int) {
        val book = bookList[position]
        holder.titleView.text = book.title
        holder.authorView.text = book.author
        holder.descriptionView.text = book.description
        holder.imageView.setImageResource(book.imageResource)
    }

    override fun getItemCount(): Int {
        return bookList.size
    }

    class BookViewHolder(bookView: View) : RecyclerView.ViewHolder(bookView) {
        val titleView: TextView = bookView.findViewById(R.id.title)
        val authorView: TextView = bookView.findViewById(R.id.author)
        val descriptionView: TextView = bookView.findViewById(R.id.description)
        val imageView: ImageView = bookView.findViewById(R.id.image)
    }
}
